package-res/cdo/complex.js

```diff
@@ -176,6 +179,39 @@ def
         return index ? (keySep + atom.key) : atom.key;
     },
 
+    /**
+     * Checks if a given atom is only defined at the root proto object.
+     *
+     * @returns {Boolean} True if the atom is defined at the root proto object and
+     *                    it is not overshadowed by any more specific proto.
+     */
+    isAtomRootDefault: function(atom) {
+        var root = def.rootProtoOf(this.atoms);
```

Comment:
Ideally we would traverse the prototype chain only once. Looking at the code of `def.rootProtoOf`, though, I can see why you do it.